docs/cub/developer_overview.rst

```diff
@@ -594,6 +594,151 @@ All the kernel does is derive the proper policy type,
 unwrap the policy to initialize the agent and call one of its ``Consume`` / ``Process`` functions.
 Agents hold kernel bodies and are frequently reused by unrelated device-scope algorithms.
 
+To gain a better understanding of why we use ``MaxPolicy`` instead of
+``ActivePolicy`` to instantiate the kernel, consider the following example which
+minimally reproduces the CUB dispatch layer.
+
+.. code-block:: c++
+
+    #include <cuda/std/type_traits>
+    #include <cstdio>
+
+    /// In device code, CUB_PTX_ARCH expands to the PTX version for which we are
+    /// compiling. In host code, CUB_PTX_ARCH's value is implementation defined.
+    #ifndef CUB_PTX_ARCH
+    #  if !defined(__CUDA_ARCH__)
+    #    define CUB_PTX_ARCH 0
+    #  else
+    #    define CUB_PTX_ARCH __CUDA_ARCH__
+    #  endif
+    #endif
```

Comment:
I would just explain `CUB_PTX_ARCH` but not include the definition

Reply:
My intention was to have compilable code so that the user could reproduce the results, but if you think that is not necessary I can remove this